src/core/infer_request.cc

```diff
@@ -707,6 +707,18 @@ InferenceRequest::Normalize()
               ", got " + DimsListToString(input.OriginalShape()));
     }
 
+    // Must make sure input shape is specified which is not checked in
+    // compare function above
+    for (const auto& dim : input.OriginalShape()) {
```

Comment:
Instead of iterating over the shape again can we just replace CompareDimsWithWildcard above with a local compare function that does exactly what we want and gives appropriate error messages.